paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/kafka/KafkaLogStoreRegister.java

```diff
@@ -82,7 +83,12 @@ public KafkaLogStoreRegister(LogStoreTableFactory.RegisterContext context) {
         } else {
             this.timeout = Duration.parse(context.getOptions().get(REGISTER_TIMEOUT.key()));
         }
-        this.partition = context.getOptions().get(LOG_SYSTEM_PARTITIONS);
+
+        this.partition =
+                context.getOptions().getOptional(LOG_SYSTEM_PARTITIONS).isPresent()
+                        ? context.getOptions().get(LOG_SYSTEM_PARTITIONS)
+                        : context.getOptions().get(BUCKET);
```

Comment:
Bucket may be -1 according to https://cwiki.apache.org/confluence/display/PAIMON/PIP-6%3A+Unaware-Bucket+Table